FrontEnd/src/components/ProfilePage/FormComponents/FormFields/HalfFormField.js

```diff
@@ -13,17 +20,33 @@ const HalfFormField = (props) => {
                 </label>
             </div>
             <div className={css['fields__field']}>
-                <input
-                    type={props.inputType ? props.inputType : 'text'}
-                    className={`${css['fields__field--input']} ${props.name === 'email' && css['disabled__field']}`}
-                    name={props.name}
-                    value={props.value}
-                    placeholder={props.fieldPlaceholder ? props.fieldPlaceholder : 'Введіть текст'}
-                    onChange={props.updateHandler}
-                    required={(props.requredField) ? 'required' : ''}
-                    disabled={(props.name === 'email') ? 'disabled' : ''}
-                    maxLength={props.maxLength}
-                />
+                {shouldShowTooltip ? (
+                    <Tooltip title={props.value} placement="bottom">
+                        <input
+                            type={props.inputType || 'text'}
+                            className={`${css['fields__field--input']} ${props.name === 'email' && css['disabled__field']}`}
+                            name={props.name}
+                            value={truncatedEmail}
+                            placeholder={props.fieldPlaceholder || 'Введіть текст'}
+                            onChange={props.updateHandler}
+                            required={props.requredField ? 'required' : ''}
+                            disabled={props.name === 'email' ? 'disabled' : ''}
+                            maxLength={props.maxLength}
+                        />
+                    </Tooltip>
+                ) : (
+                    <input
+                        type={props.inputType || 'text'}
+                        className={`${css['fields__field--input']} ${props.name === 'email' && css['disabled__field']}`}
+                        name={props.name}
+                        value={truncatedEmail}
+                        placeholder={props.fieldPlaceholder || 'Введіть текст'}
+                        onChange={props.updateHandler}
+                        required={props.requredField ? 'required' : ''}
+                        disabled={props.name === 'email' ? 'disabled' : ''}
+                        maxLength={props.maxLength}
+                    />
+                )}
             </div>
```

Comment:
You can avoid conditional rendering by conditionally setting the tooltip title directly within the property:
<Tooltip title={shouldShowTooltip ? props.value : ''} placement="bottom">